exonum-java-binding/core/src/main/java/com/exonum/binding/core/storage/indices/ProofMapIndexProxy.java

```diff
@@ -151,6 +210,17 @@
     return map;
   }
 
+  private static <K> ProofMapKeyCheckingSerializerDecorator<K> decorateKeySerializer(
+      Serializer<K> keySerializer, boolean keyHashing) {
+    if (!keyHashing) {
+      ProofMapKeySizeCheckingSerializerDecorator<K> sizeCheckingSerializerDecorator =
+          ProofMapKeySizeCheckingSerializerDecorator.from(keySerializer);
+      return ProofMapKeyCheckingSerializerDecorator.from(sizeCheckingSerializerDecorator);
+    } else {
+      return ProofMapKeyCheckingSerializerDecorator.from(keySerializer);
```

Comment:
Why is this needed at all if it can return just keySerializer, with no extra decorator types?

Reply:
To check for not null values?

Reply:
There is one for that already.

Reply:
com.exonum.binding.common.serialization.CheckingSerializerDecorator